qiskit/qobj/_validation.py

```diff
@@ -7,142 +7,17 @@
-        # place.
-        schema_file_path = os.path.join(sdk, 'schemas', 'qobj_schema.json')
-        with open(schema_file_path, 'r') as schema_file:
-            _SCHEMAS['qobj'] = json.load(schema_file)
-
-    return _SCHEMAS['qobj']
-
-
-def validate_qobj_against_schema(qobj):
-    """Validates a QObj against a schema."""
-    qobj_schema = _load_qobj_schema()
-
-    try:
-        jsonschema.validate(qobj.as_dict(), qobj_schema)
-    except jsonschema.ValidationError as err:
-        newerr = QobjValidationError(
-            'Qobj failed validation. Set Qiskit log level to DEBUG for further information.')
-        newerr.__cause__ = _SummaryValidationError(err)
-        logger.debug('%s', _format_causes(err))
-        raise newerr
-
-
-def _format_causes(err, level=0):
-    """Return a cascading explanation of the validation error in the form
-    of::
-
-        <validator> failed @ <subfield_path> because of:
-            <validator> failed @ <subfield_path> because of:
-                ...
-            <validator> failed @ <subfield_path> because of:
-                ...
-            ...
-
-    For example::
-
-        'oneOf' failed @ '<root>' because of:
-            'required' failed @ '<root>.config' because of:
-                'meas_level' is a required property
-
-    Meaning the validator 'oneOf' failed while validating the whole object
-    because of the validator 'required' failing while validating the property
-    'config' because its 'meas_level' field is missing.
-
-    The cascade repeats the format "<validator> failed @ <path> because of"
-    until there are no deeper causes. In this case, the string representation
-    of the error is shown.
-
-    Args:
-        err (jsonschema.ValidationError): the instance to explain.
-        level (int): starting level of indentation for the cascade of
-            explanations.
-
-    Return:
-        str: a formatted string with the explanation of the error.
-
-    """
-    lines = []
-
-    def _print(string, offset=0):
-        lines.append(_pad(string, offset=offset))
-
-    def _pad(string, offset=0):
-        padding = '  ' * (level + offset)
-        padded_lines = [padding + line for line in string.split('\n')]
-        return '\n'.join(padded_lines)
-
-    def _format_path(path):
-        def _format(item):
-            if isinstance(item, str):
-                return '.{}'.format(item)
-
-            return '[{}]'.format(item)
-
-        return ''.join(['<root>'] + list(map(_format, path)))
-
-    _print('\'{}\' failed @ \'{}\' because of:'.format(
-        err.validator, _format_path(err.absolute_path)))
-
-    if not err.context:
-        _print(str(err.message), offset=1)
-    else:
-        for suberr in err.context:
-            lines.append(_format_causes(suberr, level+1))
-
-    return '\n'.join(lines)
+from qiskit import QISKitError, validate_json_against_schema
 
 
 class QobjValidationError(QISKitError):
```

Comment:
I think this class is no longer raised. Would you mind to remove it, please? If there is code handling this error, it should be replaced with `SchemaValidationError`, don't you think?

Reply:
This error is still raised when the QObj is validated not involving the schema. See https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/qobj/_qobj.py#L65